File: src/test/java/compiler/block/imports/ImportBlock.java
package compiler.block.imports;

public class ImportBlock extends Block {

    public String directory, fileName;

    public ImportBlock(String directory, String fileName) {
        super(null, false, false);
        this.directory = directory;
        this.fileName = fileName;
    }

    @Override
    public void init() {

    }

    public String getName() {
        return null;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return null;
    }

    @Override
    public String getOpeningCode() {
        return null;
    }

    @Override
    public String getBodyCode() {
        return null;
    }

    @Override
    public String getClosingCode() {
        return null;
    }

}

File: src/main/scala/test_classes/block/imports/ImportBlock.scala
package test_classes.block.imports

import test_classes.block.Block

/**
  * Represents an import
  *
  * @param directory
  * @param fileName
  */
class ImportBlock(var directory: String, var fileName: String) extends Block(null, false, false) {

  def init() {}

  def getName: String = ""

  def getValue: String = ""

  def getType: String = "import"

  def getOpeningCode: String = ""

  def getClosingCode: String = ""

  override def toString: String = "import: " + directory + " : " + fileName


}